File: app/src/main/java/com/example/aichat/AddedModelAdapter.java
package com.example.aichat;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.button.MaterialButton;

import java.util.ArrayList;
import java.util.List;

public class AddedModelAdapter extends RecyclerView.Adapter<AddedModelAdapter.Holder> {
    private List<ProviderInfo.ProviderModelInfo> items = new ArrayList<>();
    private OnModelActionListener listener;

    public interface OnModelActionListener {
        void onEditAlias(ProviderInfo.ProviderModelInfo model);
        void onRemove(ProviderInfo.ProviderModelInfo model);
    }

    public void setOnModelActionListener(OnModelActionListener l) {
        listener = l;
    }

    public void setModels(List<ProviderInfo.ProviderModelInfo> list) {
        items = list != null ? new ArrayList<>(list) : new ArrayList<>();
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_model_added, parent, false);
        return new Holder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder h, int position) {
        ProviderInfo.ProviderModelInfo m = items.get(position);
        String display = m.nickname != null && !m.nickname.isEmpty() ? m.nickname + " (" + m.modelId + ")" : m.modelId;
        h.modelDisplay.setText(display);
        h.btnEditAlias.setOnClickListener(v -> {
            if (listener != null) listener.onEditAlias(m);
        });
        h.btnRemove.setOnClickListener(v -> {
            if (listener != null) listener.onRemove(m);
        });
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class Holder extends RecyclerView.ViewHolder {
        TextView modelDisplay;
        MaterialButton btnEditAlias, btnRemove;

        Holder(View v) {
            super(v);
            modelDisplay = v.findViewById(R.id.modelDisplay);
            btnEditAlias = v.findViewById(R.id.btnEditAlias);
            btnRemove = v.findViewById(R.id.btnRemove);
        }
    }
}

File: app/src/main/java/com/example/aichat/AddedModelAdapter.kt
package com.example.aichat

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.button.MaterialButton
import java.util.ArrayList

class AddedModelAdapter : RecyclerView.Adapter<AddedModelAdapter.Holder>() {

    private var items: List<ProviderInfo.ProviderModelInfo> = ArrayList()
    private var listener: OnModelActionListener? = null

    interface OnModelActionListener {
        fun onEditAlias(model: ProviderInfo.ProviderModelInfo)
        fun onRemove(model: ProviderInfo.ProviderModelInfo)
    }

    fun setOnModelActionListener(l: OnModelActionListener) {
        listener = l
    }

    fun setModels(list: List<ProviderInfo.ProviderModelInfo>?) {
        items = if (list != null) ArrayList(list) else ArrayList()
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_model_added, parent, false)
        return Holder(v)
    }

    override fun onBindViewHolder(h: Holder, position: Int) {
        val m = items[position]
        val display = if (m.nickname != null && m.nickname.isNotEmpty()) "${m.nickname} (${m.modelId})" else m.modelId
        h.modelDisplay.setText(display)
        h.btnEditAlias.setOnClickListener { listener?.onEditAlias(m) }
        h.btnRemove.setOnClickListener { listener?.onRemove(m) }
    }

    override fun getItemCount(): Int = items.size

    class Holder(v: View) : RecyclerView.ViewHolder(v) {
        val modelDisplay: TextView = v.findViewById(R.id.modelDisplay)
        val btnEditAlias: MaterialButton = v.findViewById(R.id.btnEditAlias)
        val btnRemove: MaterialButton = v.findViewById(R.id.btnRemove)
    }
}
